android/app/src/main/java/com/sample/component/ComponentActivity.kt

```diff
@@ -6,25 +6,53 @@ import android.os.Bundle
 import android.view.MenuItem
 import androidx.fragment.app.Fragment
 import com.facebook.react.ReactActivity
+import com.facebook.react.ReactActivityDelegate
 
 class ComponentActivity : ReactActivity() {
 
+    class ComponentActivityDelegate(
```

Comment:
You can change this class to be `inner` then you won't need to pass `initialProperties` as a constructor property. The class should also be private:`private inner ComponentActivity: ReactActivity()`

Reply:
Scrap that. I think you can just avoid having a prop inside of `ComponentActivity` altogether by:

```kotlin
private inner class ComponentActivityDelegate(
    activity: ReactActivity,
    mainComponentName: String?,
) : ReactActivityDelegate(activity, mainComponentName) {
    override fun getLaunchOptions(): Bundle? {
      return intent.extras?.getBundle(COMPONENT_INITIAL_PROPERTIES)
    }
}
```